src/Microsoft.Health.Dicom.CosmosDb/CosmosDataStore.Retrieve.cs

```diff
@@ -3,42 +3,67 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
-//using System;
-//using System.Collections.Generic;
-//using System.Linq;
-//using System.Text;
-//using System.Threading.Tasks;
-
-//using FellowOakDicom;
-
+using Microsoft.Azure.Cosmos;
 using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Features.Query;
 
 namespace Microsoft.Health.Dicom.CosmosDb
 {
     public partial class CosmosDataStore
     {
-        private async Task<IEnumerable<DataField>> GetItems()
+        private async Task<IEnumerable<DataField>> GetItems(IReadOnlyCollection<QueryFilterCondition> filterConditions)
         {
-            //var data = new DataField() { Value = new { val = 45 } };
-            var query = "SELECT * FROM c WHERE c['value']['00080020']['Value'][0] = '20200922'";
-            //var query = "SELECT * FROM c";
-            var res1 = Container.GetItemQueryIterator<DataField>(query);
+            CosmosDbQueryGenerator cosmosDbQueryGenerator = new CosmosDbQueryGenerator();
+            var maxItemCount = 100;
+            var query = "";
+            foreach (var condition in filterConditions)
+            {
+                condition.Accept(cosmosDbQueryGenerator);
+            }
+
+            // FUTURE TODO: continuation tokens ? will need a POST for a long token
+            query = $"SELECT * FROM c WHERE {cosmosDbQueryGenerator.OutputQuery()}";
+            Console.WriteLine($"the query is : {query}");
+            // FUTURE TODO: get from URL parameter
+            string continuation = "";
+            var queryOptions = new QueryRequestOptions()
+            {
+                MaxItemCount = maxItemCount,
+            };
+            // TODO: if continuation token not null/empty, then add it to teh query iterator
+            var res1 = Container.GetItemQueryIterator<DataField>(query, requestOptions: queryOptions);
             var list = new List<DataField>();
-            while (res1.HasMoreResults)
+
+            using (res1)
             {
-                foreach (var item in await res1.ReadNextAsync())
+                while (res1.HasMoreResults)
                 {
-                    list.Add(item);
+                    FeedResponse<DataField> response = await res1.ReadNextAsync();
+                    if (response.Count > 0)
+                    {
+                        //foreach (var item in response.Resource)
```

Comment:
probably don't need this commented out in the code if the .AddRange is working as expected.